src/features/swap/SwapForm.tsx

```diff
@@ -79,11 +80,30 @@ function SwapForm() {
 function SwapFormInputs({ balances }: { balances: AccountBalances }) {
   const { address, isConnected } = useAccount()
 
-  const { values, setFieldValue } = useFormikContext<SwapFormValues>()
+  const { values, errors, setFieldValue, setErrors, validateForm, setFieldTouched } =
+    useFormikContext<SwapFormValues>()
   const { amount, direction, fromTokenId, toTokenId } = values
 
+  useEffect(() => {
+    setErrors({})
+  }, [setErrors, values])
+
   const { isLoading, quote, rate } = useSwapQuote(amount, direction, fromTokenId, toTokenId)
 
+  useEffect(() => {
+    ;(async () => {
+      if (Object.keys(errors).length !== 0) {
+        await validateForm()
+      }
+    })().catch((e) => {
+      logger.error(e)
+    })
+  }, [validateForm, errors, quote])
```

Comment:
Is this extra hook necessary or would the form's `validateOnChange={false}` option suffice? 
https://formik.org/docs/guides/validation#when-does-validation-run

Reply:
I think we do want the validation on change there could be a scenario where we are exceeding the trading limits and a new quote comes in and the new quote is below our trading limits. But not sure if that's what you mean :)

Reply:
Sorry, let me clarify: If you change the `validateOnChange` prop to `true` on the `Formik<SwapFormValues>` component higher up in this file, does that give you the behavior you want without needing an new `useEffect` hook here?

Reply:
Sorry for the confusion 😁 
I think it does exactly what I wanted 👍 
Thank you 🙏